### DISCUSSION TITLE
Berry newbie - Trivial example fails on custom build Tasmota 14.1.0 - what am I missing?

### DISCUSSION THREAD

**Discussion by mi-hol** [arendst/Tasmota#22541] [Q&A] [answered]
tried https://tasmota.github.io/docs/Berry-Cookbook/#adding-commands-to-tasmota via copy&paste from the web page but it reports a syntax error.
Simple commands work though.

Am I missing something?

![image](https://github.com/user-attachments/assets/3c8ed48a-5364-46f3-950a-f3e4bde9c499)

**Comment by barbudor** [comment]
You are adding a Tasmota command to be called from the Tasmota console, not from Berry console

**Reply by sfromis** [reply]
Commands you create can be invoked via the Tasmota console (not Berry), MQTT, serial, rules, Berry `tasmota.cmd` and so on.

**Reply by mi-hol** [reply]
Right, coming from other shells I missed the part "Then in Tasmota Console:" 
Thank you so much!

Was I the only newbie to struggle with this from my view uncommon separation of consoles?

**Reply by mi-hol** [reply]
> Was I the only newbie to struggle with this from my view uncommon separation of consoles?

https://github.com/arendst/Tasmota/discussions/22327#discussioncomment-10993602 is showing a similar confusion

**Comment by mi-hol** [comment]
@barbudor @sfromis learning a bit more I found a way to simplify the tutorial, thereby avoiding potential confusion.
Simplify via using `tasmota.cmd` instead of closing Berry Console and opening Tasmota Console.
![image](https://github.com/user-attachments/assets/e1988b35-47d4-44d3-b317-0cf51db56395)

How about changing the tutorial to only use Berry console?
I'd be happy to provide a PR if you agree to above proposal

**Reply by barbudor** [reply]
The goal of `tasmota.add_cmd()` is to enhance Tasmota with new commands that can be used from the console, mqtt, http
Adding a command and then using `tasmota.cmd()` to use it from Berry is non-sense.
When you need to call something from Berry, just use Berry functions.

The current explaination is perfectly in-line with the goal of `add_cmd()`

Tasmota commands are the core of Tasmota configuration and usage.
Berry is an add-on, like Scriptng-Language, which helps adding functionalities.

**Reply by mi-hol** [reply]
> Adding a command and then using `tasmota.cmd()` to use it from Berry is non-sense.

@barbudor I did more looking around and found that my proposed technique is actually used in at least one of the featured solutions.
Maybe you could elaborate on the reasons this approach would be "non-sense"?

**Reply by barbudor** [reply]
Where?

**Reply by barbudor** [reply]
And I already explained the purpose of 'tasmota.add_cmd'

**Reply by sfromis** [reply, marked as answer]
While a couple of the Berry cookbook examples does use `tasmota.cmd` for the purpose of executing existing Tasmota commands as part of automating something, that is a very different usage from defining your own command, and then suggesting how it technically *can* (even if not the intended way) be executed from inside of Berry. The purpose of defining something as a command, instead of a function or a Berry `module`, would generally be to be able to invoke it from outside of Berry. Like from the console, MQTT, etc.

**Reply by mi-hol** [reply]
> for the purpose of executing existing Tasmota commands as part of automating something

is my use-case, manually switching between Berry to define a command and console to run and test it, is what I would call non-sense. It seems to result in getting the best out of 2 different tools.
As so often the "correct approach" depends on several factors, there is no single right approach.
Lets leave it at that understanding.

**Reply by sfromis** [reply]
I'd say that *suggesting* to execute the command form inside Berry is actually more of a "non-sense" approach, as going to the Tasmota console serves to illustrate the point of defining a command, how it is meant to be used.
